### DISCUSSION TITLE
Support partial capture of transactions

### DISCUSSION THREAD

**Discussion by jmauzyk** [craftcms/commerce#3825] [Ideas]
I wanted to bring up the idea of adding support for partial capture of transactions in Commerce. It’s functionality that’s commonly supported by many payment gateways, and I believe it could help streamline our workflow.

Sometimes a customer places an order, but during fulfillment, we realize we can’t fulfill all the line items due to issues like stock errors or a larger B2B order taking priority. In these cases, the customer often opts to have us ship the available items.

Currently, we ship the order, capture the payment as usual, and then refund the difference for the missing items once the transaction settles. This requires extra waiting time and monitoring by our customer service team.

Being able to capture a partial amount of the transaction would simplify this process by allowing us to capture only the value of the shipped items, without needing to process refunds. It would also provide more flexibility in handling payments overall.

I’m curious if partial capture is something the team would consider for future releases. Together with the ability to void transactions, it would be a meaningful improvement for our workflows around managing payments.

Thanks for your consideration!

**Comment by lukeholder** [comment]
Thanks for the request. What payment gateway do you use? 

Stripe only supports a single capture from the initial authorization, although we could look to customize the amount there.

To charge the remaining amount later you would need to take a new payment though.

**Reply by jmauzyk** [reply]
We use Braintree, which supports multiple partial captures per authorization. In our current workflow, we don’t backorder out-of-stock items, so our primary need is to capture a portion of the original authorization based on the value of items shipped. Customers can then place a new order for any missing items when they’re back in stock.

That said, if Commerce were to support multiple partial captures from a single authorization, it could create opportunities for more streamlined handling of partial shipments and backorders. Granted, I understand this would depend on the capabilities of the gateway being used.

I noticed you mentioned that Stripe only supports a single capture per authorization. If I'm understanding the `final_capture` parameter mentioned in [the documentation](https://docs.stripe.com/api/payment_intents/capture#capture_payment_intent-final_capture) correctly, Stripe might allow for multiple partial captures, although this functionality may require configuration by Stripe support. If that's the case, functionality around multiple partial captures could benefit Stripe gateway users as well.
